### ISSUE TITLE
README files need reviewed, updated and organized.

### ISSUE THREAD

**Issue by Fish-Git** [SDL-Hercules-390/hyperion#117]
As mentioned by Drew in issue #114, Hercules's has a long list of README files:
```
[frederica:0,~/src/hyperion,03:49:26,25] ls README*
README.AIX		README.HAO		README.MSVC		README.SETUID
README.APL360		README.HDL		README.NETWORKING	README.SUN
README.BSD		README.HERCLOGO		README.OSX		README.TAPE
README.CMPSC		README.HQA		README.PTT		README.WIN64
README.COMMADPT		README.HRAF		README.REXX		README.md
README.DYNMOD		README.IOARCH		README.RUNTEST
README.ECPSVM		README.ISSUES		README.RXVT4APL
README.EXTPKG		README.MINGW		README.S37X
[frederica:0,~/src/hyperion,03:51:31,26] ls README* | wc -l
      29
```
which can be rather intimidating and confusing to someone new to Hercules or wanting to get involved in Hercules development.

Drew goes on to say:
> If a project has 29 read me files, the first file read is README.NOW / README.1st / README.md assuming it's a directory to the rest. I read the README.md, and stopped when it didn't point me any where else.
> 
> You might cut a bug to reorganize them into a directory of all *.md files with an index (with live links) in a single file at the top level.

So I am "cutting this bug" to reorganize our READMEs as per Drew's suggestion.

**Comment by gsf600y** [comment]
Fish, is this the sort of thing that needs doing:

Before: https://github.com/SDL-Hercules-390/hyperion/blob/master/README.AIX
After: https://github.com/gsf600y/hyperion/blob/testbranch/readus/README.AIX.md

If so, I'd be happy to attempt to reorganise and reformat all those README into markup.

**Comment by Fish-Git** [comment]
Looks great! Thanks!

Yes, if you could do that for all of our READMEs that would be great. Having a master index would be great too. Or maybe each readme should end or begin with a mini-index to all of the other READMEs?

The idea here is not only should all of our READMEs be converted to .md markdown format, but we should have some type of master index for them all too so the person reading them can be made aware of the availability of all of the other READMEs as well.

The original request from Drew was:

> ... I read the README.md, and stopped when it didn't point me anywhere else.
> 
> You might cut a bug to reorganize them into a directory of all *.md files with an index (with live links) in a single file at the top level.

So if you could do that too (in addition to converting them all to .md markdown) that would be great!

Thanks!

**Comment by gsf600y** [comment]
Yes, agreed.

I intended on:

- "master" readme in top level directory to include index to all other readme
- other readme moved to `/readus` directory....hmmm, a little silly, ok `/readme` directory.
- individual readme files to link back to the master
- individual readme files to have a contents (or at least the ones that are suitably long).
- where one readme refers to another, convert to live link
- convert other external references to live links
- potentially insert images/screenshots if appropriate, and locate those in `/readme/images` directory

Any content updates/reworks can come later I think, excluding blatant and simple errors I know I can correct, e.g. dead URLs that need repointing.

**Comment by Fish-Git** [comment]
Perfect! Thank you.

You don't have to worry about the directory structure if you don't want to however. That's something I can take care of myself since it also requires a repository change to go along with it.  Just converting the files to .md markdown and providing an index would be a huge step forward. I can take care of the rest.

**Comment by gsf600y** [comment]
First pass tidy up done:
https://github.com/gsf600y/hyperion/blob/testbranch/README.md
Can you have a quick look through and let me know if you have any feedback?

Need to go through and double check each one, and make additional formatting corrections/tweaks.

**Comment by Fish-Git** [comment]
> Can you have a quick look through and let me know if you have any feedback?

_**BEAUTIFUL!**_ 

Absolutely _**beautiful!**_ &nbsp; `:)))`

I am _very_ pleased! &nbsp; _THANK YOU!!_  &nbsp; _(whoever you are! You never did tell us your name!)_

&nbsp;

> Need to go through and double check each one, and make additional formatting corrections/tweaks.

A quick check by myself uncovered the following glitches:

Problems in the "Table of Contents" sections of various readme pages:

The following links on the **Hercules "External Packages"** readme page do not work:
* The "_LINUX (non-Windows)_" link.

The following links on the "**LOW-LEVEL TEST CASES**" readme page do not work:
* The "_runtest" shell script_" link.
* The "_.tst file runtest command_" link.
* The "_.tst file directives_" link.
* The "_Creating test files that can run on z/CMS as well as under Hercules_" link.

On the "**HercIFC and Hercules as setuid root programs**" readme page, there's a link for "_ToDo_" but there's no "ToDo" section.

The following links on the "**Preliminary 2703 BSC Support**" readme page do not work:
* The "_Bugs, Caveats_" link.

The following links on the "**Hercules Dynamic Modules (on Windows)**" readme page do not work:
* The "_Requires Files_" link.
* The "_Building (making)_" link.

The following links on the "**ECPS:VM & Extended VM Assists**" readme page do not work:
* The "_How to enable/disable VM Assists_" link.

On the "**Hercules Dynamic Loader**" readme page, there's an extra (duplicate) link for "_Commands_" at the end that should be removed (i.e. link 7 is a duplicate of link 2).

The following links on the "**Host Resource Access Facility**" readme page do not work:
* The "_DIAGNOSE X'F18' Subcodes_" link.

In the "Summary" section (last section at the end) of the "**Hercules Tape Support Enhancements SPE/Fixes**" page, there is mention of the "TMOUNT" sample. This should be a link to the "TMOUNT.txt" file in the 'util' subdirectory: e.g. https://github.com/SDL-Hercules-390/hyperion/blob/master/util/TMOUNT.txt

The following links on the "**Hercules Input/Output Architecture**" readme page do not work:
* The "_Motivation for I/O Architecture Enhancements_" link.
* The "_Channel Input/Output Architecture in ESA/390 or z/Architecture_" link.
* The "_Enabling Channel-based I/O in ESA/390 or z/Architecture_" link.

On the same page, there's an extra (duplicate) link for "_Architecture Feasibility_" at the end that should be removed (i.e. link 6 is a duplicate of link 3).

The following links on the "**Unresolved Issues**" readme page do not work:
* The "_I/O Subsystem_" link.

_____
All in all I am _**very**_ happy! &nbsp; `:)`

Just fix the above glitches and we should be good to go!

_(Note: I did not yet review the actual content of any of the pages in detail yet. I figure that can come later. That is to say, once we have the new readmes in place, then I (or anyone else) can then come back later and make whatever minor editorial changes that might need to be made -- if any.)_

**Comment by gsf600y** [comment]
Great, glad you're pleased!

Ah, yes....finally I have sussed out how links with special characters work, hence all those dud links you found.  Thanks for spotting those!
Corrected all the above errors.

I'd like to spend a few more days just checking through them for errors, and especially improving the consistency.
For example, I've tended to put commands as code, like this:
 -   Enter the `D A,L` command.

But I wasn't sure if I should also start putting references to files in the same way, e.g:
 -   Open the `Hercules_VS2008.sln` Solution file.

Any preference?

Regards
Patrick :)

**Comment by Fish-Git** [comment]
Hi Patrick! :)

> Any preference?

The second:

- The one with the `fixed width` command.

**Comment by Fish-Git** [comment]
> I've tended to put commands as code

Which I prefer, so thank you.

> should also start putting references to files in the same way

Nah. Leave file references as normal (variable width) font.

BUT... if, appropriate, make them links. If they're not links though, then leave them as normal font.

**Comment by Fish-Git** [comment]
(and if they _are_ links, leave them as normal font too, i.e. file references should _not_ (IMO) use fixed with fonts. Only commands and references to code/variables, etc, should be in fixed with fonts.)

**Comment by Fish-Git** [comment]
@gsf600y Patrick,

I re-reviewed everything and as far as I'm concerned we look good to go. Are you going to create a github pull request for your changes? Or should I just basically apply your changes manually myself by copying all of your .md markdown files over from your repository into SDL-Hyperion? Either way is fine by me!

**Comment by gsf600y** [comment]
Great, thanks for that.
I'll just do a bit more checking myself too, then create a pull request tonight (UK) :)

**Comment by Fish-Git** [comment]
@gsf600y Patrick:

We're trying to get the new 4.2 release sometime today (Sun 2019-07-07) so if you can manage to get your pull request ready within the next 10-12 hours or so (preferably sooner!) that would be great.

Otherwise I'm prepared to just copy over your changes from your repository over into the SDL repository and commit them myself from there (with you as the author of the commit of course! (so that you get proper credit for the changes)) if that's okay.

But I'll continue waiting for a few more hours yet just in case you get your pull request ready before then.

Cool?

**Comment by gsf600y** [comment]
Pull request done - one conflict, and being a novice on GitHub, I don't quite know how it should be resolved.  I'm guessing this is probably something you can sort in a few clicks.

Also, not sure I like what I've done here....the Pull Request "files changed" tab shows files that I wasn't expecting to be included....cast your expert eye over this very carefully please Fish!
Cheers
Patrick

**Comment by Fish-Git** [comment]
@gsf600y Patrick:

> Also, not sure I like what I've done here....the Pull Request "files changed" tab shows files that I wasn't expecting to be included....cast your expert eye over this very carefully please Fish!

Yes, I see that.  I suspect (but am not sure; I'm kind of a novice too when it comes to github pull requests) it might be because your repository is missing some recent commits. I made a few minor commits a couple of days ago that I suspect your repository doesn't have.

I _think_ (but again, am not sure) you simply need to do a "pull" on your repository to merge my recent commits into your repository. Not sure what to do after you  do that. Do you need to delete your existing pull request and create a new pull request? Or will your existing pull request be updated (adjusted) automatically? I don't know! I'm sort of new to this too!

Worst cast: withdraw (delete?) your existing pull request, do a git pull to pickup my changes, and _then_ do a brand new pull request.

Thanks.

**Comment by gsf600y** [comment]
Ok....took a few attempts, and had to download the GitHub Desktop software to sort it out, but the latest pull request (#3) I believe is now OK :)

**Comment by Fish-Git** [comment]
@gsf600y

I'm afraid I'm going to have to REJECT your pull request, Patrick.&nbsp;`:(`

Everything looks good EXCEPT ... the (probably accidental) deletion of msvc.makefile.includes\VSVERS.msvc. 

I might have accidentally deleted it from your repository myself when I reviewed your previous pull request and tried to fix things. (Sorry about that!)

Once again I'm not sure how best to fix the problem. Should you just re-add it back again? (I believe maybe yes) Or do you need to rescind (withdraw) this pull request and create a brand new pull request? (not sure but maybe)

Sorry!&nbsp;`:(`

**Comment by Fish-Git** [comment]
@gsf600y 

(p.s. I closed your pull request. If you can fix the problem just re-open your pull request again.)

**Comment by Fish-Git** [comment]
@gsf600y 

(and sorry about all this back and forth trouble we're having! I'm just as inexperienced with pull requests as you are and I apologize for my screwing you up!)

**Comment by gsf600y** [comment]
Ok, new request is in - see how that looks.
No need to apologise - it's all good fun ;)

**Comment by Fish-Git** [comment]
Accepted! THANK YOU, Patrick!&nbsp;`:))`

**Comment by gsf600y** [comment]
Yay! :)
Cool, no problem, glad I could help.

**Comment by gsf600y** [comment]
Just need to work on that cream website next ;)

![image](https://user-images.githubusercontent.com/24498661/60843715-5b164880-a1cf-11e9-939b-ec9e7858347f.png)
